### PR TITLE
Updating google cloud nio to 0.107.0

### LINKED ISSUE #5935: Strange behavior in ParallelCopyGCSDirectoryIntoHDFSSpark with NIO ver > 0.66
### Affected tool(s) or class(es)
`ParallelCopyGCSDirectoryIntoHDFSSpark`

### Affected version(s)
- [x] Latest public release version [version?]
- [x] Latest master branch as of [2019-05-13]

### Description 

`ParallelCopyGCSDirectoryIntoHDFSSpark` behaves in the following strange way:

* under __master/latest__ release, it __fails__ to copy a GCS "directory" containing __BAMs__
* under __master/latest__ release, it __successfully__ copies a GCS "directory" containing __reference__
* changing the nio lib version from 81 to 66 in `build.gradle`, it  __successfully__ copies GCS "directories" containing __reference__ or __BAMs__
* see attached logs

#### Steps to reproduce

Both scripts referred to below need to be updated accordingly, but trivially

* from the master branch, run the attached `test.nio.ver.81.sh`.

* branch out from master, change the literal `81` to `66` on line 69 in `build.gradle`, run the attached `test.nio.ver.66.sh`.

#### Expected behavior

Files in the "directories" given in the gs path copied successfully.

#### Actual behavior
Fail. See logs attached.

-------------
[test.nio.paraCopyHDFSSpark.zip](https://github.com/broadinstitute/gatk/files/3174143/test.nio.paraCopyHDFSSpark.zip)

UPDATE:
reuploaded attachment

### PR DESCRIPTION
**Pull request by lbergelson**
* Updating google-cloud-nio   0.81.0-alpha:shaded -> 0.100.0-alpha:shaded
* Fixes #5935

### CODE REVIEW

**Comment by lbergelson** [comment]
@SHuang-Broad Lets see if it passes tests...

**Comment by lbergelson** [comment]
Hmn. This is failing with 403 unauthorized errors.  Seems like something about authentication changed.

**Comment by SHuang-Broad** [comment]
you mean tests?
I'm running the "failed" tool on an actual dataproc cluster right now. Should have a result in 10 minutes.

**Comment by lbergelson** [comment]
Yeah, the cloud tests are all failing.

**Comment by SHuang-Broad** [comment]
@lbergelson 
Tested working.

```bash
#!/bin/bash

set -eu

echo "=============================="
export CRAM_BUCKET="gs://broad-dsde-methods-shuang/tmp/test/"
gsutil ls  -l  -h "${CRAM_BUCKET}"
echo "=============================="

# on Louis branch containing NIO 100
echo "=============================="
export GATK_DIR="/Users/shuang/GATK/gatk"
export CLUSTER_NAME="shuang-nio-100"
cd "${GATK_DIR}" && git pull && git checkout lb_update_nio && \
./gradlew clean installAll && \
cd scripts/sv/ && \
bash create_cluster.sh \
  "${GATK_DIR}" \
  broad-dsde-methods \
  "${CLUSTER_NAME}" \
  3h \
  120m \
  gs://broad-dsde-methods-sv/reference/GRCh38/ \
  "${CRAM_BUCKET}" \
  gs://broad-dsde-methods-shuang/init/default_init.sh
echo "=============================="

# on master (expected to fail)
echo "=============================="
export GATK_DIR="/Users/shuang/GATK/forks/GatkFork"
export CLUSTER_NAME="shuang-nio-81"
cd "${GATK_DIR}" && git checkout master && git fetch upstream && git rebase upstream/master && \
./gradlew clean installAll && \
cd scripts/sv/ && \
bash create_cluster.sh \
  "${GATK_DIR}" \
  broad-dsde-methods \
  "${CLUSTER_NAME}" \
  3h \
  120m \
  gs://broad-dsde-methods-sv/reference/GRCh38/ \
  "${CRAM_BUCKET}" \
  gs://broad-dsde-methods-shuang/init/default_init.sh
echo "=============================="
```

**Comment by lbergelson** [comment]
We're seeing a lot of failures of the form:

```
  com.google.cloud.storage.StorageException: 806222273987-uilktks3j6i7962rp0v7nusveer58497@developer.gserviceaccount.com does not have serviceusage.services.use access to project 685190392835.
        Caused by:
        shaded.cloud_nio.com.google.api.client.googleapis.json.GoogleJsonResponseException: 403 Forbidden
        {
          "code" : 403,
          "errors" : [ {
            "domain" : "global",
            "message" : "806222273987-uilktks3j6i7962rp0v7nusveer58497@developer.gserviceaccount.com does not have serviceusage.services.use access to project 685190392835.",
            "reason" : "forbidden"
          } ],
          "message" : "806222273987-uilktks3j6i7962rp0v7nusveer58497@developer.gserviceaccount.com does not have serviceusage.services.use access to project 685190392835."
        }
```

It looks like it now requires some new permission for the service accounts but our existing service account doesn't have that permission.

**Comment by lbergelson** [comment]
@SHuang-Broad Thanks, that's good to know that it actually fixes the problem you're having...  I think we need to investigate this error a bit more though before we can merge.

**Comment by SHuang-Broad** [comment]
Ha? That's strange.
Is the project id `685190392835` dsde-methods?

**Comment by lbergelson** [comment]
No... it's not.  It doesn't seem to correspond to any broad project I can see.

**Comment by SHuang-Broad** [comment]
@lbergelson I did an experiment in #6046 with NIO 94, same errors.

**Comment by SHuang-Broad** [comment]
@lbergelson 

I've run a few more tests, and found the following table

```
NIO_VER	403_DENIED_PROJECT_ID
99	685190392835
98	685190392835
97	685190392835
96	539774316296
95	685190392835
94	685190392835
93	539774316296
92	685190392835
91	539774316296
90	539774316296
89	PASS
88	PASS
87	PASS
86	PASS
85	PASS
```
Googling both of the two mysterious project IDs, I landed onto tests by this Fiji project ([here](https://travis-ci.com/seomoz/fiji/jobs/43151327) and [here](https://travis-ci.com/seomoz/fiji/jobs/43513954)).
I parsed the test log and found the relevant part (for ID `685190392835`)

```
"gce":{

    "instance":{
        "attributes":{
            "startup-script":"#!/usr/bin/env bash\necho poweroff | at now + 130 minutes\ncat > ~travis/.ssh/authorized_keys <<EOF\nssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDdzVnIEg2ribEvhEvFjR9IFPAkIVtQwZhlgUAHu1BgjBugFRiqg3eaPMOeOuIZBvzwoyotHIVp3XvAfivGyCW4Ke7+2cqlcX1L8kcmoWLm2fdLGlLr/lZnAjQtexMC76uLtR8udqWA0e2sqrSJs4H/blOQmHWPrl/VSG7daoVptzqXihRmXN+/Huo7mTxAjTUEjk4IOBn7sv7G5qLrEPv78AJIZhWHdhUTGLvx+YpzQvX8pE53TMi9W4ovkZTCwhSO3WYyBOY7H1xjeYb9XWTeP563Du1b0JMpQgtFLQUVXio9NzXZE55ovvGDRSLds+VfPsv4G/Whhq76dEZ+wZO3\n\nEOF\n"
        },
        "cpuPlatform":"Intel Haswell",
        "description":"Travis CI python test VM",
        "disks":[{"deviceName":"persistent-disk-0","index":0,"mode":"READ_WRITE","type":"PERSISTENT"}],
        "hostname":"testing-gce-ec8614d2-40a2-4138-801e-d42d811590a2.c.travis-ci-prod-2.internal",
        "id":8221730359445041428,
        "image":"",
        "licenses":[{"id":"1000010"}],
        "machineType":"projects/685190392835/machineTypes/n1-standard-2",
        "maintenanceEvent":"NONE",
        "networkInterfaces":[{"accessConfigs":[{"externalIp":"104.198.203.242","type":"ONE_TO_ONE_NAT"}],"forwardedIps":[],"ip":"10.128.0.163","network":"projects/685190392835/networks/default"}],
        "scheduling":{"automaticRestart":"TRUE","onHostMaintenance":"MIGRATE","preemptible":"FALSE"},
        "serviceAccounts":{
            "685190392835-compute@developer.gserviceaccount.com":{
                "aliases":["default"],
                "email":"685190392835-compute@developer.gserviceaccount.com",
                "scopes":["https://www.googleapis.com/auth/userinfo.email",
                          "https://www.googleapis.com/auth/devstorage.full_control",
                          "https://www.googleapis.com/auth/compute"]
            },
            "default":{
                "aliases":["default"],
                "email":"685190392835-compute@developer.gserviceaccount.com",
                "scopes":["https://www.googleapis.com/auth/userinfo.email",
                          "https://www.googleapis.com/auth/devstorage.full_control",
                          "https://www.googleapis.com/auth/compute"]}
            },
        "tags":["testing"],
        "virtualClock":{"driftToken":"11704388862566216373"},
        "zone":"projects/685190392835/zones/us-central1-b"
    },

    "project":{
        "attributes":{
            "sshKeys":"henrik:ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQChY0pdGXohYN7KRnQa3VIcDoVBrxZVHkhOFc1SROV2T+gTOunYbOW5C4V1P2MGG6FcKeoQTJzXgPbZurM5l1AfEbKeCde778QyyxbcjpYvKyY5b4qVO79nOKAg1qHIqUl+2txv7X6tPv4Q99T7UBechuc5awnkJZKqP1s1qJ9BYYYAPukZPbhAkjkvPSaJfIi+py2p6L9mXFtrAhYNH1flE9GErAsf2Hq8zQvx4hmTseumv4Fb9rVogcBJOqhmDQmYwTg2rEbdLAjbqY7Sf4kjdOfF7uhwasZgVMjF1z5utnvHd2wC/cjkuDZB4UhetLTeOWDtvgZxF/uVJSTU2AGD google-ssh {\"userName\":\"henrik@travis-ci.org\",\"expireOn\":\"2016-03-04T00:52:57+0000\"}\nhenrik:ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBJ6IOlU4vY6QLWKOX52Opcdx/2zNgJyMq7ntIf8qD+CbMMfUy5C6WJnjn4E2lvYqYaVIotY196cVazh0Jj8E/co= google-ssh {\"userName\":\"henrik@travis-ci.org\",\"expireOn\":\"2016-03-04T00:52:42+0000\"}\n"
        },
        "numericProjectId":685190392835,
        "projectId":"travis-ci-prod-2"
    }
},


```

In the meantime, I've also found [this site](https://github.com/travis-ci/build-environment-updates/blob/master/2017Q1/gce/connie-trusty/node-attributes.yml.diff) for ID `539774316296`.

Not sure if these info are useful for you guys (who know much more about this than I do) to debug the issue.

**Comment by lbergelson** [comment]
I finally opened an issue with google cloud to look into this... https://github.com/googleapis/google-cloud-java/issues/5884

**Comment by droazen** [comment]
@lbergelson As discussed, it's not essential that we update to 107. I'd be happy with either 105 or 106, since the shading fix was released in 105.

File: build.gradle [hunk 1]
```diff
@@ -59,7 +59,7 @@ repositories {
 
 final requiredJavaVersion = "8"
 final htsjdkVersion = System.getProperty('htsjdk.version','2.20.3')
-final picardVersion = System.getProperty('picard.version','2.20.5')
+final picardVersion = System.getProperty('picard.version','2.20.6-5-g1b4178f-SNAPSHOT')
```
Comment (droazen) [inline build.gradle]: Update to actual released version of Picard

**Pull request merged**